store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java

```diff
@@ -0,0 +1,335 @@
+                LOGGER.warn("For the best performance you should set the spark config 'spark.serializer' = 'org.apache.spark.serializer.KryoSerializer'");
+            }
+            final String sparkKryoRegistrator = conf.get("spark.kryo.registrator", null);
+            if (sparkKryoRegistrator == null || !sparkKryoRegistrator.equals("uk.gov.gchq.gaffer.spark.serialisation.kryo.Registrator")) {
+                LOGGER.warn("For the best performance you should set the spark config 'spark.kryo.registrator' = 'uk.gov.gchq.gaffer.spark.serialisation.kryo.Registrator'");
+            }
+            return super.createContext(user);
+        } else {
+            LOGGER.info("Setting up the Spark session using " + this.getProperties().getSparkMaster() + " as the master URL");
+            final SparkSession spark = SparkSession.builder()
+                    .appName("Gaffer Parquet Store")
+                    .master(this.getProperties().getSparkMaster())
+                    .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+                    .config("spark.kryo.registrator", "uk.gov.gchq.gaffer.spark.serialisation.kryo.Registrator")
+                    .getOrCreate();
+            return super.createContext(new SparkUser(user, spark));
+        }
+    }
+
+    protected void addAdditionalOperationHandlers() {
+        addOperationHandler(GetDataFrameOfElements.class, new GetDataframeOfElementsHandler());
+        addOperationHandler(ImportRDDOfElements.class, new AddElementsFromRDDHandler());
+    }
+
+    @Override
+    protected OutputOperationHandler<GetElements, CloseableIterable<? extends Element>> getGetElementsHandler() {
+        return new GetElementsHandler();
+    }
+
+    @Override
+    protected GetAllElementsHandler getGetAllElementsHandler() {
+        return new GetAllElementsHandler();
+    }
+
+    @Override
+    protected DummyGetAdjacentEntitySeedsHandler getAdjacentIdsHandler() {
+        return new DummyGetAdjacentEntitySeedsHandler();
+    }
+
+    protected OperationHandler<? extends AddElements> getAddElementsHandler() {
+        return new AddElementsHandler();
+    }
+
+    @Override
+    protected Class<? extends Serialiser> getRequiredParentSerialiserClass() {
+        return Serialiser.class;
+    }
+
+    @Override
+    protected Object doUnhandledOperation(final Operation operation, final Context context) {
+        throw new UnsupportedOperationException("Operation " + operation.getClass() + " is not supported by the ParquetStore.");
+    }
+
+    @SuppressFBWarnings(value = "BC_UNCONFIRMED_CAST_OF_RETURN_VALUE", justification = "The properties should always be ParquetStoreProperties")
+    @Override
+    public ParquetStoreProperties getProperties() {
+        return (ParquetStoreProperties) super.getProperties();
+    }
+
+    public void loadIndices() throws StoreException {
+        try {
+            if (this.groupToIndex == null) {
```

Comment:
I think you could get concurrent modification exceptions with this groupToIndex map. An alternative might be to instead of clearing the map, to create a new map then assign the new map to groupToIndex at the end of the method.